src/odemis/acq/stitching/test/registrar_test.py

```diff
@@ -0,0 +1,295 @@
+        [tiles,pos] = decomposeImage(img,overlap,numTiles)
+        
+        registrar = ShiftRegistrar()
+        for i in range(len(pos)):
+            registrar.addTile(tiles[i])
+            calculatedPositions = registrar.getPositions()[0]
+            self.assertAlmostEqual(calculatedPositions[i][0],pos[i][0],places=1)
+            self.assertAlmostEqual(calculatedPositions[i][1],pos[i][1],places=1)
+
+    def test5(self):
+        """ Test acquisition in vertical direction """
+        img = Image.open("images/test2.tiff")
+        numTiles = 2
+        overlap = 0.2
+        [tiles,pos] = decomposeImage(img,overlap,numTiles,"verticalLines")
+        
+        registrar = ShiftRegistrar()
+        for i in range(len(pos)):
+            registrar.addTile(tiles[i])
+            calculatedPositions = registrar.getPositions()[0]
+            
+            self.assertAlmostEqual(calculatedPositions[i][0],pos[i][0],places=1)
+            self.assertAlmostEqual(calculatedPositions[i][1],pos[i][1],places=1)
+            
+    def test6(self):
+        """ Test acquisition in horizontal zigzag direction  """
+        img = Image.open("images/test2.tiff")
+        numTiles = 2
+        overlap = 0.2
+        [tiles,pos] = decomposeImage(img,overlap,numTiles,"horizontalZigzag")
+        
+        registrar = ShiftRegistrar()
+        for i in range(len(pos)):
+            registrar.addTile(tiles[i])
+            calculatedPositions = registrar.getPositions()[0]
+            self.assertAlmostEqual(calculatedPositions[i][0],pos[i][0],places=1)
+            self.assertAlmostEqual(calculatedPositions[i][1],pos[i][1],places=1)
+                
+    def test7(self):
+        """ Test case not generated by decompose.py file and manually cropped """
+        img = Image.open("images/test.png")
+        cropped1 = img.crop((0,0,400,400))
+        cropped2 = img.crop((322,4,722,404))
+  
+        registrar = ShiftRegistrar()
+        tile1 = model.DataArray(np.array(cropped1.convert("L")),{
+                model.MD_PIXEL_SIZE: [1/20,1/20],  # m/px
+                model.MD_POS: (200/20, img.size[1]/20-200/20),  # m
+            })
+        tile2 = model.DataArray(np.array(cropped2.convert("L")),{
+                model.MD_PIXEL_SIZE: [1/20,1/20],  # m/px
+                model.MD_POS: (520/20, img.size[1]/20-200/20),  # m
+            })
+        registrar.addTile(tile1)
+        registrar.addTile(tile2)
+        calculatedPositions = registrar.getPositions()[0]
+        self.assertAlmostEqual(calculatedPositions[1][0],522/20,places=1)
+        self.assertAlmostEqual(calculatedPositions[1][1],img.size[1]/20-204/20,places=1)          
+        
+def decomposeImage(img, overlap=0.1, numTiles = 5, method="horizontalLines"):
+    """ 
+    Decomposes image into tiles for testing. The tiles overlap and their center positions are subject to random noise.
+    Returns list of tiles and list of the actual positions. 
+    img: square image obtained by PIL module
```

Comment:
Please, change the img argument to take a 2D numpy array instead. The convention in Odemis is to keep image in that format. The main gotcha is that the shape in numpy is ordered YX, while in PIL, it's XY.